### PR TITLE
refactor: Use ESLint for linting.

### PR DESCRIPTION
**Pull request by ntwb**

### CODE REVIEW

File: lib/configs/ava.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
 module.exports = {
 	env: {
-		es6: true
+		es6: true,
```
Comment (GaryJones) [inline lib/configs/ava.js:3]: I saw the dangling comma commits go through. This is now acceptable now then?

Comment (ntwb) [reply lib/configs/ava.js:3]: Yes, details and link to the Gutenberg discussion in #79

File: package.json [hunk 2]
```diff
@@ -55,7 +55,7 @@
   "devDependencies": {
     "ava": "~0.19.1",
     "coveralls": "~2.13.1",
-    "eslint": "~4.1.0",
+    "eslint": "~4.1.1",
```
Comment (GaryJones) [inline package.json]: Any reason for not using `"^4.1"`

Comment (ntwb) [reply package.json]: Yes, per https://github.com/eslint/eslint#semantic-versioning-policy:

> _"According to our policy, any minor update may report more errors than the previous release (ex: from a bug fix). As such, we recommend using the tilde (~) in package.json e.g. "eslint": "~3.1.0" to guarantee the results of your builds."_

stylelint is also adopting this same policy for the upcoming v8.

**Pull request merged**